Compatibility with CMake < 3.5 has been removed from CMake

Hi,

Trying to build the [AUR package](https://aur.archlinux.org/packages/nemu) I got the following error:

```
==> Starting build()...
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!
==> ERROR: A failure occurred in build().
    Aborting...
 -> error making: nemu-exit status 4
```

I can confirm that changing `cmake_minimum_required(VERSION 2.8.12)` to `cmake_minimum_required(VERSION 3.6)` in the first line of `CMakeLists.txt` allows the build to complete and the program works.

Thanks

**edit**: Arch is on cmake 4.0.1 right now

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.10)
 
 project(nemu C)
 

